### ISSUE TITLE
undefined method `split' for nil:NilClass

### ISSUE THREAD

**Issue by charri** [bitrise-steplib/steps-nunit-runner#1]
When I add nunit to the workflow I receive the following error:

`NUnit test failed: undefined method `split' for nil:NilClass` (ruby error)

I guess the error must be in here as its the only ruby file using split: [builder.rb](https://github.com/bitrise-steplib/steps-nunit-runner/blob/master/xamarin-builder/builder.rb)

Configuration: Any CPU | Release

**Comment by hvaughan3** [comment]
I am also seeing this issue. Wish I knew ruby to try and fix it, though maybe it is a problem with my solution. I only see the split statements in the `analyzer.rb` file.

**Comment by viktorbenei** [comment]
Hi,

Thanks for reporting, we're working on fixing the issue & to generate a better error message!

**Comment by hvaughan3** [comment]
@viktorbenei Great, I really appreciate the help.

**Comment by godrei** [comment]
Hi,

tomorrow we will release a new version, however the fix is ready in beta version.
If someone wants to use this beta version:

- On bitrise.io navigate to Workflow -> MANAGE WORKFLOW -> bitrise.yml
- Replace the `nunit-runner` step to the beta version  
  `nunit-runner` looks like in bitrise.yml:  
```
...
    - nunit-runner@0.9.0:
...
```
  change this to:
```
...
    - git::https://github.com/godrei/steps-xamarin-test-cloud-for-android.git@fix:
...
```

After this, nunit runner step is not loaded from the steplib, but from the repo which we will share tomorrow into the steplib.

**Comment by godrei** [comment]
sorry i updated my answer `https://github.com/godrei/steps-xamarin-test-cloud-for-android.git` -> `https://github.com/godrei/steps-nunit-runner.git`

**Comment by mleatherb** [comment]
Plugged in the forked version from godrei and it errors out. 

Build command failed
--- Stack trace: ---
#<RuntimeError: Build command failed>
["/private/var/folders/lb/8n5bn9k975qgw662jpqdy7mm0000gn/T/bitrise006271240/step_src/xamarin-builder/builder.rb:49:in `build_solution'", "/var/folders/lb/8n5bn9k975qgw662jpqdy7mm0000gn/T/bitrise006271240/step_src/step.rb:69:in `<main>'"]

**Comment by hvaughan3** [comment]
@godrei 
I also tried running the beta version and received the following error:

    undefined method `each_with_index' for #<String:0x007fb95a859a88>
    --- Stack trace: ---
    #<NoMethodError: undefined method `each_with_index' for #<String:0x007fb95a859a88>>
    ["/private/var/folders/lb/8n5bn9k975qgw662jpqdy7mm0000gn/T/bitrise737776827/step_src/xamarin-builder/builder.rb:87:in `run_nunit_tests'", "/var/folders/lb/8n5bn9k975qgw662jpqdy7mm0000gn/T/bitrise737776827/step_src/step.rb:72:in `<main>'"]
    Build failed

**Comment by godrei** [comment]
solved in [v0.9.1](https://github.com/bitrise-steplib/steps-nunit-runner/releases/tag/0.9.1)
